### PR TITLE
Changed from classification to category

### PR DESCRIPTION
**Pull request by LeeSingChanYau**
Changed “Classification” to “Category” in task edit view.
How to test:
Admin Login → Other Links → Projects → Click WBS Icon by project chosen → Choose WBS → Add or Edit Task
Make sure that the field name is category and not classification

### CODE REVIEW

**Comment by one-community** [comment]
Excellent, thank you Alan. Reminder to change the status to "Done" and add the PR # and link to the Bugs doc whenever you post your PRs

File: src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx [hunk 1]
```diff
@@ -353,17 +352,15 @@ const AddTaskModal = props => {
     if (props.level >= 1) {
       const classificationMother = props.tasks.taskItems.find(({ _id }) => _id === props.taskId)
         .classification;
-        if(classificationMother){
-          setClassification(classificationMother);
-        }
-    }
-    else {
+      if (classificationMother) {
```
Comment (PrabhjotSembhi) [inline src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx:355]: I suggest renaming the variable too. 
Please rename all classifications and its related variables to "categories"

File: src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx [hunk 2]
```diff
@@ -606,11 +603,15 @@ const AddTaskModal = props => {
                 </td>
               </tr>
               <tr>
-                <td scope="col">Classification</td>
+                <td scope="col">Category</td>
                 <td scope="col">
-                  <select value={classification} onChange={e => setClassification(e.target.value)} >
-                    {classificationOptions.map(cla =>{
-                      return <option value={cla.value} key={cla.value}>{cla.label}</option>
+                  <select value={classification} onChange={e => setClassification(e.target.value)}>
```
Comment (PrabhjotSembhi) [inline src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx:608]: Everything seems fine. just rename variables.

**Review by PrabhjotSembhi — COMMENTED** [review]
I have checked it. working fine. 

Please rename variables also. that would be great. 

You can see my review video here. 

https://user-images.githubusercontent.com/82662768/215276899-252ce472-50a0-4eeb-8dcb-71b1727bc924.mp4

**Comment by RafaelCastilhoCastro** [comment]
Hi @LeeSingChanYau,

I tested the PR and the fields are renamed to Category in both Add Task and Edit Task forms.
As observed by @PrabhjotSembhi I'd recommend changing the variables names for consistency's sake.
Thanks.

**Comment by AnishPandita** [comment]
Hey @LeeSingChanYau I went over the Pr and everything seems to work as intended.  
The code structure seems reasonable and well oriented. As suggested by others renaming the variables to something more appropriate would be nice.

https://user-images.githubusercontent.com/111321945/215285386-293a2d6c-4118-4d13-811b-ce1a2eb94272.mov

**Comment by Gukal7** [comment]
Tested the changes, everything working perfect. Like others said, variable name changes would be great in terms of code readability. 
Although, I would suggest if the sub categories could be clearly identifiable, like the row line between 3.8 and 4 could have more width/boldness, else it's difficult to understand what lies under what category apart from seeing the # number. Or maybe changing the color scheme for the sub categories.
 
Screenshot 2023-01-28 at 3 15 26 PM

**Comment by Filipe-Oliv** [comment]
Hi @LeeSingChanYau

I tested your PR and its focus is fine.

As a suggestion for other PR, as @Gukal7 told you, I think the subs are not clearly defined. You could change the font color of the subs, for example, or something that makes it clear that the directory is a sub, like a branch symbol next to the folder name.

[week 03 - 02-07-23 (6).webm](https://user-images.githubusercontent.com/114119668/217363156-d2dccb50-cd2b-48bc-a0ed-9b96dba4c0d5.webm)

**Comment by LeeSingChanYau** [comment]
Hey @Filipe-Oliv @Gukal7, I made some changes to the PR. As requested I changed all variable names from classification to category. And also changed the task name that is displayed as subcategory. If a task is a subcategory like 1.1, it would display something like "Energy/Save-Energy". I am using backslash to be clear what is the parent category. Please let me know if this looks good.
image

**Review by PrabhjotSembhi — COMMENTED** [review]
This is good now

**Comment by one-community** [comment]
If good, someone please approve this and then I'll move it to the Management review stage.

**Review by Filipe-Oliv — DISMISSED** [review]
All right. Looks good.

**Comment by RafaelCastilhoCastro** [comment]
@one-community Jae, this PR is approved.

**Comment by one-community** [comment]
Thank you. Moving to Management review

**Comment by agujjar2206** [comment]
Thanks for submitting this PR! I tested the changes locally by checking out the branch and running the application, and everything appears to be working correctly for both Add and Edit Task. I did not encounter any issues during testing.

Classification to Category Add
Classification to Category Edit

**Review by agujjar2206 — DISMISSED** [review]

**Review by PrabhjotSembhi — APPROVED** [review]

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx [hunk 3]
```diff
@@ -25,6 +25,7 @@ const Task = props => {
   const userPermissions = props.state.auth.user?.permissions?.frontPermissions;
   useEffect(() => {
     setIsCopied(false);
+    console.log('props.filteredTasks:', props.filteredTasks);
```
Comment (rauleffting) [inline src/components/Projects/WBS/WBSDetail/Task/Task.jsx:28]: Please, delete this debug output.

**Review by rauleffting — CHANGES_REQUESTED** [review]
It is working perfectly. However, a debug output was left on the code. Please delete. After that, I'll approve. Thank you very much!

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx (hunk 3 above)
Comment (LeeSingChanYau) [reply src/components/Projects/WBS/WBSDetail/Task/Task.jsx]: @rauleffting thanks for the feedback! Fixed it, please review again.

**Comment by Lunaliar** [comment]
@LeeSingChanYau this works on my end! 
I also have some housekeeping if you want to take care of any of it while we're working on these files.

In addTaskModal.jsx
1. lines 114 & 116 there are debug functions
2. lines 636, 660 & 684 have a second className prop
3. line 730 has two onClicks on the same component

In AddTaskModal.jsx and EditTaskModal.jsx onKeypress is depreciated and could be swapped to onKeyDown

In ImportTask.jsx and Task.jsx the props for the components could be spread for less redundancy

**Review by suigunfjc — APPROVED** [review]
Hi @LeeSingChanYau
I pulled and tested your codes, everything looks pretty good to me and I will approve it. Good Job!

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx (hunk 3 above)
Comment (PrabhjotSembhi) [reply src/components/Projects/WBS/WBSDetail/Task/Task.jsx]: @rauleffting Hi, can you please check If your requested changes have been made, please approve this. 
Thanks.

**Comment by ayush0904** [comment]
Excellent work!!! I have tested it and it's working perfect.
A small suggestion that could be implemented in future. We can have a constant file where we can save the dropdown values and labels, so that for updating the labels and values to multiple places we only have to change the constant file.

**Review by tingedudu8 — APPROVED** [review]
Hi @LeeSingChanYau, everything looks good to me. The name has been changed, as well as the variable names. The subtasks' names has been ajusted to make the file structure much clearer. Thanks!
Screen Shot 2023-02-21 at 6 32 57 PM

File: src/components/Projects/WBS/WBSDetail/Task/Task.jsx (hunk 3 above)
Comment (LeeSingChanYau) [reply src/components/Projects/WBS/WBSDetail/Task/Task.jsx]: @one-community @rauleffting I deleted the console.log.

**Pull request merged**